### PR TITLE
feat: add web platform

### LINKED ISSUE #769: [Feature]: Add Support for Web Bluetooth W3C standard ( preliminary )
### FlutterBluePlus Version

all

### Flutter Version

all

### What OS?

All

### OS Version

Supported Chrome

### Bluetooth Module

any

### What is your feature request?

With Chrome ( behind flag ) supporting Web Bluetooth standard, This can be a great addition for Flutter Web Development.

If anyone is interested please let me know

### Logs

```shell
Chrome Web BLE support:
https://developer.chrome.com/docs/capabilities/bluetooth?hl=en

Web Bluetooth W3C Spec:
https://webbluetoothcg.github.io/web-bluetooth/
```

### PR DESCRIPTION
**Pull request by tnc1997**
Closes #769

### CODE REVIEW

**Comment by chipweinberger** [comment]
sdk: ">=2.15.1 <4.0.0"
  sdk: "^3.4.0"

is this change required? why

**Comment by tnc1997** [comment]
I think that this change is necessary because [extension types](https://dart.dev/language/extension-types), which are used by [JS interop](https://dart.dev/interop/js-interop/usage#interop-type-members), were fully introduced in version 3.3.1 of the Dart SDK.

**Comment by chipweinberger** [comment]
oh, also please test the example app.

**Comment by chipweinberger** [comment]
> if (!kIsWeb && Platform.isAndroid) 

do we need both?

**Comment by chipweinberger** [comment]
I see your adding a lot of checks for adapterState

on web, can we not detect adapterState?

lets just default it to 'on' then.

File: README.md [hunk 1]
```diff
@@ -571,6 +571,10 @@ Make sure you have granted access to the Bluetooth hardware:
 
 <img width="528" alt="Screenshot 2023-12-11 at 10 32 04 AM" src="https://github.com/boskokg/flutter_blue_plus/assets/1863934/554079ef-4627-4dfc-97e3-1f07f84a0f3c">
 
+### Enable the Bluetooth API
```
[the remark this answers is not in the document]
Comment (tnc1997) [reply README.md]: https://github.com/boskokg/flutter_blue_plus/pull/964/commits/68ea7f63a516fcc21ad9e987667ef68266730787

**Comment by tnc1997** [comment]
> > if (!kIsWeb && Platform.isAndroid)
> 
> do we need both?

Yes I think so, because `Platform` is not supported on web, because it is provided by `dart:io`.

**Comment by tnc1997** [comment]
> I see your adding a lot of checks for adapterState
> 
> on web, can we not detect adapterState?
> 
> lets just default it to 'on' then so users dont need to add so many checks.

Sure I will push a change to remove the exception from `getAdapterState` and return `BmAdapterStateEnum.on`.

**Comment by tnc1997** [comment]
https://github.com/boskokg/flutter_blue_plus/pull/969#issuecomment-2294330693

**Pull request closed without merging**